Updates coding guidelines (#489)
haiku/website: content/development/coding-guidelines.md

{
	...
}

> //  Empty functions converted to single line functions

Reviewer: Should this be allowed outside of the class declaration?

I think this is fine:

```
class Example
{
    void EmptyFunction() {}
};
```

But this isn't:

```
void Example::EmptyFunction() {}
```

In the case it's outside the class declaration it should instead be:

```
void
Example::EmptyFunction()
{
}
```

The case where an empty function is defined outside a class should be quite uncommon anyway. We would usually put it directly in the class declaration.

	...
}

> // Empty functions in class should be converted to inline functions

Reviewer: "inline" means something specific in C and C++, to avoid confusion we should not use that word here.

```suggestion
// Empty functions in class definitions can be written on a single line
```

class Example
{

>     void FooFunction() {}

Reviewer: Even if it is in a single line, the alignment of the various columns should be preserved.
Also you had spaces instead of tabs for indentation.

```suggestion
		void			FooFunction() {}
```

};

> // Empty functions defined outside class should not be converted to inline functions

Reviewer: ```suggestion
// Empty functions defined outside class must be formatted the same as other functions
```
